kubeflow/jupyter/ui/default/template.html

```diff
@@ -1,110 +1,233 @@
+            <input class="form-control" id='memory' name='memory'
+                   placeholder='100Mi, 1.5Gi, etc' required>
+            </input>
+          </div>
+          <p class="text-muted" style="padding: 10px;">
+            Specify the total amount of RAM reserved by your Notebook
+            (e.g. <span><code>2.0Gi</code></span>).
+          </p>
+        </div>
+      {% endblock %}
 
-  {% block memory %}
-  <!-- Memory -->
-  <div class="panel panel-primary">
-    <div class="panel-heading">
-      <i class="fas fa-memory"></i>
-      <label>Memory</label>
-    </div>
-    <div class="panel-body" style="padding: 10px;">
-      <input class="form-control" id='memory' name='memory'
-             placeholder='100Mi, 1.5Gi, etc'>
-      </input>
-    </div>
-    <p class="text-muted" style="padding: 10px;">
-      Specify the total amount of RAM reserved by your Notebook
-      (e.g. <span><code>2.0Gi</code></span>).
-    </p>
-  </div>
-  {% endblock %}
+      {% block workspaceVolume %}
+        <!-- Workspace Volume -->
+        <div class="panel panel-primary">
+          <div class="panel-heading">
+            <i class="fas fa-laptop-code" style="padding: 0px 2px;"></i>
+            <label>Workspace Volume</label>
+          </div>
+          <div class="panel-body" id="workspace_volume">
+              <div class="col-sm-2" style="width: 14%"><label>Type</label></div>
+              <div class="col-sm-3" style="width: 22%"><label>Name</label></div>
+              <div class="col-sm-2" style="width: 12%"><label>Size (Gi)</label></div>
+              <div class="col-sm-3" style="width: 29%"><label>Mount Path</label></div>
+              <div class="col-sm-2" style="width: 14%"><label>Access Mode</label></div>
+              <div class="col-sm-*">
+              <div class="col-sm-2" style="width: 14%">
+                <select class="form-control" name="ws_type" id="ws_type">
+                  <option selected>New</option>
+                  <option>Existing</option>
+                </select>
+              </div>
+              <div class="col-sm-3" style="width: 22%">
+                <input class="form-control" name="ws_name" id="ws_name"
+                       list="suggest_pvcs_disabled" required>
+                </input>
+              </div>
+              <div class="col-sm-2" style="width: 12%">
+                <input class="form-control" name="ws_size" id="ws_size" placeholder='10'
+                       type="number" step="0.5" min="0" required>
+                </input>
+              </div>
+              <div class="col-sm-3" style="width: 29%">
+                <input class="form-control" id="ws_mount_path" name="ws_mount_path" required></input>
+              </div>
+              <div class="col-sm-2" style="width: 14%">
+                <select class="form-control" id="ws_access_modes" name="ws_access_modes" required>
+                  <option value="ReadWriteOnce">RWO</option>
```

Comment:
should there be a tooltip or alt text for these?

Reply:
Hi Kam,

we have already added a tooltip for the Access Mode `<select>` elements that shows up when the Type of the Volume is set to 'Existing`, to inform about autocompletion. As such, instead of a second tooltip, we suggest that the available options of  the Access Mode dropwdown menu should be updated as follows :
- RWO --> ReadWriteOnce
- RWX --> ReadWriteMany
- ROX --> ReadOnlyMany